docs/upgrading.md

```diff
@@ -4,6 +4,28 @@ description: Learn how to upgrade Next.js.
 
 # Upgrade Guide
 
+## Upgrading from version 11 to 12
+
+#### `--dir` flag is no longer supported in `next lint`
```

Comment:
Can this change be made in a backwards compatible way?

Reply:
[Re-included the `--dir` flag](https://github.com/vercel/next.js/pull/26777/files#diff-14d870abad761e1d36cbfd875474ffd988f452e3328e8ca41e58c219fc22bd5dR143) so that it still works and remains backwards compatible 👍 

I don't know if we could do the same with regards to adding `--base-dir` as a brand new flag however, as the new behavior will match the ESLint CLI